pkg/services/ngalert/state/manager.go

```diff
@@ -248,17 +262,32 @@ func translateInstanceState(state ngModels.InstanceStateType) eval.State {
 	}
 }
 
-func (st *Manager) annotateState(ctx context.Context, alertRule *ngModels.AlertRule, labels data.Labels, evaluatedAt time.Time, state eval.State, previousState eval.State) {
-	st.log.Debug("alert state changed creating annotation", "alertRuleUID", alertRule.UID, "newState", state.String(), "oldState", previousState.String())
+// This struct provides grouping of state with reason, and string formatting.
+type InstanceStateAndReason struct {
+	State  eval.State
+	Reason string
+}
+
+func (i InstanceStateAndReason) String() string {
+	r := fmt.Sprintf("%v", i.State)
+	// We never want to write down (Normal) or (Alerting)
+	if len(i.Reason) > 1 && i.State.String() != i.Reason && i.Reason != eval.Alerting.String() {
+		r = r + fmt.Sprintf(" (%v)", i.Reason)
+	}
```

Comment:
this can be simplified after changes you did. I think this code should not decide whether include reason.

Reply:
You're right - we already decide it now by not setting the reason.